### PR TITLE
Remote Control Robots [WIP]

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
A new type of mob that others can assume control of

-If the pilot takes damage or is otherwise attacked, restrained, etc, they get kicked back to their body

-Alt clicking yourself as the robot returns you to your body

-If the robot dies, you return to your body

Still to do: Sprites, a tool to control them (you just alt click to take over right now, as a placeholder), different variants (surveillance, bomb, etc).

### CODE REVIEW

**Comment by GunHog** [comment]
Watch_Dogs 2!

**Comment by SgtxRootbeer** [comment]
Can this work with the current bots (mule, beepsky, medibot, etc?)
If not can we get variants that serve similar purposes?
Also, can we get a bot that is capable of doing construction? If people having hardsuits is an issue with power creep, a construction bot could be given to every engineer instead.

**Comment by FantasticFwoosh** [comment]
New beepsky skin for bartender RC ROBOT RAGE CAGE confirmed

**Comment by TehZombehz** [comment]
Nanotrasen tech support, now wirelessly dragging a medbot to you to tell you to stop dieing.

**Comment by FantasticFwoosh** [comment]
Idea directly from IRC

> **Hatticus_Finch	>malf ai shunting to a pAI unit and using its mobile mode to run away**

**Comment by anconfuzedrock** [comment]
DEPLOYING DRONE

**Comment by Anonmare** [comment]
>yfw you're a HoP and someone drives an RC car with a TTV attached to it under your table

**Comment by ShadowDeath6** [comment]
This could be used for AI remote control shell, minus the whole getting hurt part.

**Comment by KorPhaeron** [comment]
If the AI is under attack it'd probably be happy to get bumped back to its core to defend as well

**Comment by Anonmare** [comment]
Generally if the AI is being attacked directly, it can't do much anyway

**Comment by AdamElTablawy** [comment]
this better fucking include the toy mechs

**Comment by DrPillzRedux** [comment]
Right now it's just going to be a unique little RC drone. I'd imagine if an admin wants to be funny they can just change the sprite to something else.

Ian drone.

**Comment by Greenpoison** [comment]
I really like this idea, I hope it can be implemented into cargo nicely and possibly take away the risk of hacked mules.

**Comment by calzilla1** [comment]
Bot crafting when

**Comment by kevinz000** [comment]
add borgs for adminspawned one :^)

**Comment by DrPillzRedux** [comment]
Honestly this could be the start of something beautiful. Someone could use this system and make a RC car cage match thing, with saws on them. Holodeck derby matches. Emag holodeck and chop legs off with saw drone.

**Comment by FantasticFwoosh** [comment]
Mmm, what about giving mining robots a function in this? If you dont want to have to put up with the sometimes troublesome pathing AI or you dont specifically have the mining job while cushily on the station go do it by remote.

@DrPillzRedux you mean a LMG SAW drone https://tgstation13.org/phpBB/viewtopic.php?f=9&t=8760&p=232878#p232878

**Comment by JJRcop** [comment]
https://github.com/tgstation/tgstation/blob/9c3ed16ec1eb9ef9bfdad4d68dfd4fa030540e68/code/modules/mob/mob_defines.dm#L138

**Comment by optimumtact** [comment]
KOrrrrrrrr

**Comment by KorPhaeron** [comment]
I've added remotes and icons for the RC car.

Now I just have to make the various types of remote controlled robots themselves and it will be ready.

File: code/modules/mob/living/simple_animal/hostile/remote_control.dm [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	initial_pilot_health = null
+
+/mob/living/simple_animal/hostile/remote_control/proc/assume_control(mob/living/user)
+	if(pilot || !requires_pilot)
+		user << "It's already being piloted."
+	else
+		initial_pilot_health = user.health
+		pilot = user
+		if(pilot.mind)
+			pilot.mind.transfer_to(src)
```
Comment (JJRcop) [inline code/modules/mob/living/simple_animal/hostile/remote_control.dm]: Actually putting them into the bot is definitely really iffy to me, especially when we already have a remote_control var on mobs.

Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/remote_control.dm]: Does the remote_control var let you attack, use powers etc?

Comment (JJRcop) [inline code/modules/mob/living/simple_animal/hostile/remote_control.dm]: I didn't think of that, you're right.

**Comment by optimumtact** [comment]
Stale

**Comment by KorPhaeron** [comment]
hmmm

**Comment by SgtxRootbeer** [comment]
Awww

**Comment by GunHog** [comment]
Please do not toy with my heart, Kor!

**Pull request closed without merging**